src/tasks/FileUploadTask/Interfaces/IUploadEventHandlers.ts

```diff
@@ -0,0 +1,16 @@
+/* eslint-disable @typescript-eslint/type-annotation-spacing */
+import { Range } from "../Range";
+
+/**
+ * Interface enabling progress handling with callbacks.
+ */
+export interface UploadEventHandlers {
+	/**
+	 * Parameters that are passed into the progress, completed, failure callback options.
+	 */
+	extraCallbackParam?: unknown;
+	/**
+	 * Callback function called on each slice upload during the LargeFileUploadTask.upload() process
+	 */
+	progress?: (range?: Range, extraCallbackParam?: unknown) => void;
```

Comment:
```suggestion
	progress?: (range?: Range, extraCallbackParam?: () => void) => void;
```

Reply:
This will remain unknown. extraCallbackParam can be anything that the user wishes to access in the progress call back

Reply:
sure, but when our code invokes it, it invokes it with a fixes number of parameters, and doesn't "get" any returned value right?
Typing this parameter would "help" subscribers by letting them know "you can expect all those parameters to be passed to you, and you're not expected to return anything"

Reply:
progress?: (range?: Range, extraCallbackParam?: unknown) " => void " . The part in quotes suggests that there is no return.

I can pass anything as extraCallbackParam and not only a callback.

Reply:
aaaah I get it, this is for them to pass something to themselves, correct?

Why do we need it at all then? if we removed it, the protoype here would be `(range?: Range) => void)`

And then if they want to pass things to themselves, they can do that during the assignment `(range) => (range, param1, param2, param3) => { //body }`

Reply:
I am not sure if TypeScript allows this. I will take a look at this in a separate PR if possible.